jdisc_http_service/src/main/java/com/yahoo/jdisc/http/server/jetty/HealthCheckProxyHandler.java

```diff
@@ -124,21 +115,86 @@ protected void doStop() throws Exception {
         super.doStop();
     }
 
+    private static class ProxyRequestTask implements Runnable {
+
+        final AsyncContext asyncContext;
+        final ProxyTarget target;
+        final HttpServletResponse servletResponse;
+        final ServletOutputStream output;
+
+        ProxyRequestTask(AsyncContext asyncContext, ProxyTarget target, HttpServletResponse servletResponse, ServletOutputStream output) {
+            this.asyncContext = asyncContext;
+            this.target = target;
+            this.servletResponse = servletResponse;
+            this.output = output;
+        }
+
+        @Override
+        public void run() {
+            StatusResponse statusResponse = target.requestStatusHtml();
+            servletResponse.setStatus(statusResponse.statusCode);
+            if (statusResponse.contentType != null) {
+                servletResponse.setHeader("Content-Type", statusResponse.contentType);
+            }
+            output.setWriteListener(new WriteListener() {
+                @Override
+                public void onWritePossible() throws IOException {
+                    if (output.isReady()) {
+                        if (statusResponse.content != null) {
+                            output.write(statusResponse.content);
+                        }
+                        asyncContext.complete();
```

Comment:
I don't like the spec here. _How much_ can you write without blocking? Generally, this should be a `while` loop, but since you only do a single write here, and then close, I guess this is fine.

Reply:
Yes, it's a bad interface. The servlet has no clue on how much data to produce. This method typically implemented as a `while` loop if the servlet had multiple byte buffers to write.